src/test/java/org/example/bookshop/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,319 @@
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private DataSource dataSource;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        sqlCommandExecutor(dataSource, "books/delete-books-with-category.sql");
+    }
+
+    @AfterEach
+    public void afterEach() {
+        sqlCommandExecutor(dataSource, "books/delete-books-with-category.sql");
+    }
+
+    @SneakyThrows
+    static void sqlCommandExecutor(DataSource dataSource, String path) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(path));
+        }
+    }
+
+    @Test
+    @Sql(scripts = "classpath:categories/add-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", authorities = {"USER"})
+    @DisplayName("""
+            Should return list of CategoryDto categories in DB
+            """)
+    void getAll_WithDefaultPage_Success() throws Exception {
+        //Given
+        CategoryDto categoryDto1 = new CategoryDto();
+        categoryDto1.setId(1L);
+        categoryDto1.setName("lol info1");
+        categoryDto1.setDescription("some description1");
+
+        List<CategoryDto> expected = new ArrayList<>();
+        expected.add(categoryDto1);
+
+        CategoryDto categoryDto2 = new CategoryDto();
+        categoryDto2.setId(2L);
+        categoryDto2.setName("lol info2");
+        categoryDto2.setDescription("some description2");
+        expected.add(categoryDto2);
+
+        CategoryDto categoryDto3 = new CategoryDto();
+        categoryDto3.setId(3L);
+        categoryDto3.setName("lol info3");
+        categoryDto3.setDescription("some description3");
+        expected.add(categoryDto3);
+
+        CategoryDto categoryDto4 = new CategoryDto();
+        categoryDto4.setId(4L);
+        categoryDto4.setName("lol info4");
```

Comment:
better move it to the private method or better util class and simplify your test cases